File: prid1920-g13/Models/ModelsEntity/Event.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;

namespace prid_1819_g13.Models
{
public enum AccessType
{
    Public,
    Friends,
    ParticularFriend
}
    public class Event
    {
        [Key]
        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public DateTime Start_date {get;set;}
        public DateTime End_date {get;set;}
        public string Langue {get;set;}
        public int NbUsers{ get; set; }
        public AccessType AccessType { get; set;}
        public int CreatedByUserId { get; set;}
        public User CreatedByUser {get;set;}
        [NotMapped]
        public IList<User> Participants {get;set;}
        public int GameId {get;set;}
        [NotMapped]
        public Game Game {get;set;}
    }
}

File: prid1920-g13/ClientApp/src/app/models/Event.ts
import { Game } from "./Game";
import { User } from "./User";

export enum AccessType {
    Public,
    Friends,
    ParticularFriend
}

export class Event
{
    id: number;
    name: string
    description: string
    start_date: Date
    end_date: Date
    langue: string
    nbUsers: string
    accessType: AccessType
    createdByUserId: number;
    gameId: number;
    game: Game;
    participants: User[];
    constructor(data) {
        if (data) {
                this.id = data.id,
                this.name = data.name,
                this.description = data.description,
                this.start_date = data.start_date,
                this.end_date = data.end_date,
                this.langue = data.langue,
                this.nbUsers = data.nbUsers,
                this.accessType = data.accessType,
                this.createdByUserId = data.createdByUserId,
                this.gameId = data.gameId,
                this.game = data.game,
                this.participants = data.participants
        }
    }
}